sd-dreambooth-workflow/db-workflow-event-binding.yaml

```diff
@@ -0,0 +1,39 @@
+apiVersion: argoproj.io/v1alpha1
+kind: WorkflowEventBinding
+metadata:
+  name: db-finetune-event-binding
+spec:
+  event:
+    selector: discriminator == "db-finetune"
+  submit:
+    workflowTemplateRef:
+      name: db-finetune-template
+    arguments:
+      parameters:
+        - name: run_name
+          valueFrom:
+            event: payload.run_name
+        - name: instance_dataset
+          valueFrom:
+            event: payload.instance_dataset
+        - name: instance_prompt
+          valueFrom:
+            event: payload.instance_prompt
+        - name: class_dataset
+          valueFrom:
+            event: payload.class_dataset
+        - name: class_prompt
+          valueFrom:
+            event: payload.class_prompt
+        - name: output
+          valueFrom:
+            event: payload.output
+        - name: num_class_images
+          valueFrom:
+            event: "payload.num_class_images == null ? 100: payload.num_class_images"
+        - name: run_inference
+          valueFrom:
+            event: "payload.run_inference == null ? true : payload.run_inference"
+        - name: inference_only
+          valueFrom:
+            event: "payload.inference_only == null ? false : payload.inference_only"
```

Comment:
Once the `WorkflowTemplate` and `WorkflowEventBinding` are deployed, you can trigger a workflow run with a webhook like the following:
```bash
curl --location 'https://<Argo deployment>/api/v1/events/<namespace>/db-finetune' \
--header 'Authorization: Bearer <REDACTED>' \
--header 'Content-Type: application/json' \
--data '{
    "run_name": "event-test",
    "instance_dataset": "data/example-dog",
    "instance_prompt": "a photo of sks dog",
    "class_dataset": "generic/dogs-2",
    "class_prompt": "a photo of dog",
    "num_class_image": 10,
    "output": "finetunes/event-test"
}'
```